backend/src/v2/driver/driver.go

```diff
@@ -678,13 +678,12 @@ func resolveInputs(ctx context.Context, dag *metadata.DAG, iterationIndex *int,
 				}
 			}
 			value, hasValue := inputs.GetParameterValues()[name]
+
+			// Skip when hasValue == false, which means this parameter does not have an input value
 			if !hasValue {
```

Comment:
Don't you still need the isOptional check? "no value and not optional" should throw an error.

Reply:
Thank you, I added this check back.